### PR TITLE
Fix out-of-bounds in 'combine' texture modifier

### PR DESCRIPTION
**Pull request by Niklp09**
Fixes https://github.com/mt-mods/signs_lib/issues/24
TenPlus1 asked me to create this PR since github doesn't let him.

Tested, and works. But I'm not sure if this is the best approach :thinking: .

### CODE REVIEW

**Comment by tenplus1** [comment]
@Niklp09 - Thanks for the assist :)  I find that it generates signs quicker this way although anyone using their own custom font sets will need to use the convert.sh I've included to negate the image files so they work with the new setup.

**Comment by wsor4035** [comment]
> But I'm not sure if this is the best approach

agree on this, from just skimming the pr

**Comment by tenplus1** [comment]
@Niklp09 + @wsor4035 - Does something worry you guys about the fix ?

**Comment by appgurueu** [comment]
FWIW, you could "negate" the image files using the `^[invert` texture modifier to avoid messing with the media files.

**Comment by tenplus1** [comment]
@appgurueu - I did try to [invert medi but it never worked properly on transparent areas and the unicode textures ended up with artifacts, so in the end editing the files had to be done by hand or script.

**Comment by tenplus1** [comment]
If it worries anyone how the changes affect signs_lib, check out Xanadu server, we've been using it since the commit and it's working well.

**Comment by appgurueu** [comment]
I wouldn't mind the texture changes much. However, please at least run something like pngcrush - the new textures are much larger than they need to be. (You could also consider converting the textures to TGA to shave off some more bytes, or packing them up in a tilesheet.)

**Comment by tenplus1** [comment]
@appgurueu - Run optipng on every texture, please find attached the final mod in .zip form.
[signs_lib_NEW.zip](https://github.com/mt-mods/signs_lib/files/14584633/signs_lib_NEW.zip)

**Comment by Niklp09** [comment]
Commited to the branch, sorry for the delay.

**Comment by appgurueu** [comment]
Anything stopping this?

**Review by Niklp09 — COMMENTED** [review]
(can't approve my own pr) Works fine in my test world, no warnings in the client log.
Media size 1.054 MB -> 1.073 MB is ok for me too.
Since this is a bit controversial, I would like to get a second approval.

**Comment by OgelGames** [comment]
After some trial and error (the code is terrible to read), I figured out a simpler way to fix the issue: #27

File: api.lua [hunk 1]
```diff
@@ -467,15 +467,6 @@ signs_lib.charwidth_wide32 = build_char_db(32)
 
 local math_max = math.max
 
-local function fill_line(x, y, w, c, font_size, colorbgw)
-	c = c or "0"
-	local tex = { }
-	for xx = 0, math.max(0, w), colorbgw do
-		table.insert(tex, (":%d,%d=signs_lib_color_"..font_size.."px_%s.png"):format(x + xx, y, c))
```
Comment (appgurueu) [inline api.lua:474]: Aren't the `signs_lib_color` files unused now that you switched this to a color lookup table & `^[colorize`? Can't you remove them now (or is something else using them)?

**Comment by OgelGames** [comment]
I just tested this and it doesn't fix the issue, there are still out-of-bounds textures when there is text overflow.

**Comment by OgelGames** [comment]
Merged #27, closing this.

**Comment by tenplus1** [comment]
@OgelGames - Could you re-open this as a working alternative.

**Comment by S-S-X** [comment]
For now restored branch if someone is still working with it. Related issue was reopened so I guess meta discussion could go there.

**Comment by OgelGames** [comment]
> Could you re-open this as a working alternative.

As I noted previously, this PR doesn't work either. I think a lot more is needed to fix it properly.

**Comment by wsor4035** [comment]
cross posting

frankly replacing colorized textures with `^[colorize` is a good idea 
as to the implementation, didnt exactly dig to deep into it, so perhaps it had some flaw, but the concept is good

**Comment by appgurueu** [comment]
If there was a good specification (and tests?) such that we could ensure low risk of breaking compatibility, I would be inclined to suggest a rewrite of this part of the logic: Generating text strings via `[combine` is relatively simple.

Is there a good specification though?

**Comment by tenplus1** [comment]
> > Could you re-open this as a working alternative.
> 
> As I noted previously, this PR doesn't work either. I think a lot more is needed to fix it properly.

How so ? I've been using this on Xanadu and signs work fine.

**Comment by OgelGames** [comment]
> How so ? I've been using this on Xanadu and signs work fine.

It doesn't fix the out-of-bounds errors when the text overflows past the edge of the sign.

**Pull request closed without merging**